src/main/java/uk/nhs/hee/trainee/details/service/ProgrammeMembershipService.java

```diff
@@ -343,6 +358,46 @@ public boolean isPilot2024(String traineeTisId, String programmeMembershipId) {
         .equalsIgnoreCase("Oral and maxillo-facial surgery (run through)"));
   }
 
+  /**
+   * Assess if the programme membership for a trainee is in the 2024 pilot rollout. Hopefully a
+   * temporary kludge.
+   *
+   * @param traineeTisId          The TIS id of the trainee.
+   * @param programmeMembershipId The ID of the programme membership to assess.
+   * @return True, or False if the programme membership is not in the 2024 pilot rollout.
+   */
+  public boolean isPilotRollout2024(String traineeTisId, String programmeMembershipId) {
+    TraineeProfile traineeProfile = getProfileWithTssProgrammeMemberships(traineeTisId);
+
+    if (traineeProfile == null) {
+      log.info("2024 pilot rollout: [false] trainee profile {} not found", traineeTisId);
+      return false;
+    }
+
+    ProgrammeMembership programmeMembership = getCandidateProgrammeMembership(
+        traineeProfile.getProgrammeMemberships(), programmeMembershipId);
+    if (programmeMembership == null) {
+      log.info("2024 pilot rollout: [false] programme membership {} does not exist, is non-medical "
+              + "or has wrong type",
+          programmeMembershipId);
+      return false;
+    }
+
+    String managingDeanery = Objects.toString(programmeMembership.getManagingDeanery(), "");
+    LocalDate startDate = programmeMembership.getStartDate();
+    if (startDate == null) {
+      log.info("2024 pilot rollout: [false] start date is null for {}", programmeMembershipId);
+      return false;
+    }
+
+    LocalDate notificationEpoch = managingDeanery.equalsIgnoreCase("Thames Valley")
+        ? LocalDate.of(2025, 1, 31)
+        : LocalDate.of(2024, 7, 31);
```

Comment:
This date is 'overridden' in the notification service to avoid sending anything when the start date is pre-1 Nov 2024